### PR TITLE
feat: add sampleCollapseLevel option

### LINKED ISSUE #479: Auto expand the examples response
I apologize if this has been answered but couldn't find a clear answer.  I see the expand-response="200,201" re-doc option we can add which works great to expand the schema but what I need is to have the examples response expanded by default as shown in the attached image.  I need the full example response showing.  Is there a setting I can enable for this?

image

### PR DESCRIPTION
**Pull request by alex-karo**
This feature resolves #479
I added "sampleCollapseLevel" option, that set level of collapsing in response sample
Possible positive integer values, or "all"

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -242,6 +242,7 @@ You can use all of the following options with standalone version on <redoc> tag
 * `hideDownloadButton` - do not show "Download" spec button. **THIS DOESN'T MAKE YOUR SPEC PRIVATE**, it just hides the button.
 * `disableSearch` - disable search indexing and search box
 * `onlyRequiredInSamples` - shows only required fields in request samples.
+* `sampleCollapseLevel` - set the collapse level in response samples. Special value 'all' expand all response. Default value = 2.
```
Comment (RomanHotsiy) [inline README.md]:
```suggestion
* `sampleCollapseLevel` - set the default expand level for JSON payload samples (responses and request body). Special value 'all' expands all levels. The default value is `2`.
```

File: src/utils/jsonToHtml.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
 let level = 1;
-const COLLAPSE_LEVEL = 2;
+let collapseLevel;
```
Comment (RomanHotsiy) [inline src/utils/jsonToHtml.ts]: Get rid of this global variable. Use `maxCollapseLevel` in functions `arrayToHTML` and `objectToHTML`. They have the `maxCollapseLevel` in scope.

File: README.md (hunk 1 above)
Comment (RomanHotsiy) [inline README.md]: I would also rename the option to `jsonSampleExpandLevel`

File: src/utils/jsonToHtml.ts (hunk 2 above)
Comment (alex-karo) [reply src/utils/jsonToHtml.ts]: Should I provide `maxCollapseLevel` via arguments into these functions?

File: src/services/RedocNormalizedOptions.ts [hunk 3]
```diff
@@ -110,6 +111,16 @@ export class RedocNormalizedOptions {
     return value;
   }
 
+  private static normalizeSampleCollapseLevel(level?: number | string | 'all'): number {
```
Comment (RomanHotsiy) [inline src/services/RedocNormalizedOptions.ts]: You forgot to rename this to `normalizeSampleExpandLevel`.

```suggestion
  private static normalizeSampleExpandLevel(level?: number | string | 'all'): number {
```

File: src/services/RedocNormalizedOptions.ts [hunk 4]
```diff
@@ -156,6 +168,9 @@ export class RedocNormalizedOptions {
     this.onlyRequiredInSamples = argValueToBoolean(raw.onlyRequiredInSamples);
     this.showExtensions = RedocNormalizedOptions.normalizeShowExtensions(raw.showExtensions);
     this.hideSingleRequestSampleTab = argValueToBoolean(raw.hideSingleRequestSampleTab);
+    this.jsonSampleExpandLevel = RedocNormalizedOptions.normalizeSampleCollapseLevel(
```
Comment (RomanHotsiy) [inline src/services/RedocNormalizedOptions.ts]:
```suggestion
    this.jsonSampleExpandLevel = RedocNormalizedOptions.normalizeSampleExpandLevel(
```

File: src/services/RedocNormalizedOptions.ts (hunk 3 above)
Comment (alex-karo) [reply src/services/RedocNormalizedOptions.ts]: Thanks, fixed

File: src/services/RedocNormalizedOptions.ts (hunk 4 above)
Comment (alex-karo) [reply src/services/RedocNormalizedOptions.ts]: Done

**Comment by RomanHotsiy** [comment]
Thanks! 👍

**Pull request merged**